docs/DeveloperGuide.adoc

```diff
@@ -353,6 +353,60 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+* Traits that characterize the bicycle, such as `brand`, `model`, `top speed`, `wheel size`, `weight`, `material` etc.
+
+[NOTE]
+Why two lists to track loans, instead of collating them into one big list? Because each list has a specific category of loans and it is more common to encounter situations that deal with only one of the lists (e.g. checking transaction history, or checking for clashes with reservations), rather than both.
+
+Every `Loan` will have a reference to exactly one `Bike`: the bicycle that is loaned out. Conversely, every `Bike` has a reference to a list of `Loan`s, which track the bicycle's loan history and future reservations.
+
+[NOTE]
+This makes the `AddressBook`'s list of loans a little harder to manage, due to the extra references. However, this has been done anyway as referencing a bicycle's history is a very useful and frequently used operation. To manage the deletion of loans, the loan will be dereferenced from from the Bike it references before it gets deleted. The sequence diagram below demonstrates this process:
+image::DeleteLoanSequenceDiagram.png[width="800"]
+
+[NOTE]
+Due to the dependency of `Loan`s on `Bike`s, any bicycle that has been loaned out at least once will never be deleted from the system, merely archived and labelled `Out of Service`. The `Bike` can still be deleted completely if all of the `Loan`s associated with it are deleted (i.e. invalidated) first.
+
+Using this framework, it is now more efficient to implement and use a set of mini-features and operations to facilitate bicycle management, as detailed in the following sections.
+
+==== Visible bicycle statuses
+The current availability of bicycles are tracked by the system. Bicycles can thus be filtered by whether they are available or not, and the available ones are readily visible in the UI.
+
+==== Invalid loans
+Loaning out a particular bicycle during the time which it is unavailable or has already been reserved will be denied. This prevents accidents in loaning out bicycles.
+
+* When attempting to loan out a bicycle, the associated bicycle is retrieved from the system.
+[NOTE]
+The loan will be rejected if no bicycle is found that matches the bicycle name provided.
+
+* If the loan will take effect immediately, the bicycle's status is checked. If the bicycle is unavailable, the loan is rejected, and the bicycle's status is reported to the user.
+* If the loan is a reservation, i.e. it schedules a loan for the future, the bicycle's future reservations are checked. LoanBook will attempt to insert the reservation into the list. If the reservation clashes with the one directly before or after it, the loan is rejected, and the clash is reported to the user.
+
+==== Bicycles for particular needs
+The list of bicycles can be filtered using the same algorithm to filter the list of loans in order to find bicycles with particular properties, to suit particular customer's needs or activities.
+
+==== Design Considerations
+
```

Comment:
Should this be filled?

Reply:
No, it directly leads to the subsections below